src/rust/uk.mangainua/src/lib.rs

```
@@ -5,11 +5,35 @@ mod helper;
 use aidoku::{
 	error::Result,
 	prelude::*,
+	std::html::Node,
 	std::net::Request,
 	std::{net::HttpMethod, String, Vec},
 	Chapter, Filter, FilterType, Manga, MangaContentRating, MangaPageResult, MangaStatus,
 	MangaViewer, Page,
 };
+use core::cmp::Ordering;
+
+fn parse_user_hash_and_query(document: Node) -> (String, String) {
+	let mut script_data = String::new();
+	for script in document.select("script").array() {
+		let node = script.as_node().expect("Script not a node");
+		let text = node.outer_html().read();
+		if text.contains("site_login_hash") {
+			script_data = text;
+			break;
+		}
+	}
+
+	let user_hash = script_data
+		.split("site_login_hash = '")
+		.nth(1)
+		.and_then(|s| s.split('\'').next())
+		.expect("Failed to parse user hash");
```

this should return an error. don't use `expect`.